src/lib/profiles/data-management/Current/LoggingManagement.cpp

```diff
@@ -2146,6 +2246,171 @@ void CircularEventReader::Init(CircularEventBuffer * inBuf)
     }
 }
 
+WEAVE_ERROR CircularEventBuffer::SerializeEvents(TLVWriter & writer)
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+
+    TLVType container;
+
+    uint8_t *inBufStart = mBuffer.QueueHead();
+    size_t inBufLen = mBuffer.DataLength();
+    uint8_t *bufStart = mBuffer.GetQueue();
+    size_t bufLen = mBuffer.GetQueueSize();
+    uint8_t *eventData = new uint8_t[inBufLen];
```

Comment:
We should not need to allocate here, and it could be a local array variable, right?

Reply:
I think so, I blv the event buffer itself is a local array.